openhtf/util/measurements.py

```diff
@@ -138,6 +138,19 @@ def WithValidator(self, validator):
     self.validators.append(validator)
     return self
 
+  def WithArgs(self, **kwargs):
+    """Creates a new Measurement, see openhtf.PhaseInfo.WithArgs."""
+    new_meas = mutablerecords.CopyRecord(self)
+    if '{' in new_meas.name:
+      formatter = lambda x: x.format(**kwargs)
+    else:
+      # str % {'a': 1} is harmless if str doesn't use any interpolation.
```

Comment:
let's actually enforce that measurement names only contain alphanumeric and underscore, and don't start with a number.  this way you're guaranteed to be able to set measurements via normal attribute access instead of having to use `setattr` because you used some stupid character.